File: backends/python-fastapi/app/usecase/auth/dto.py
"""
認証ユースケース Dto モジュール。

Author: Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
"""
from dataclasses import dataclass
from typing import Optional


@dataclass
class AuthLoginDto:
    """OAuth ログイン Dto。"""

    provider: int
    provider_id: str
    name: str
    email: str
    avatar: Optional[str] = None


@dataclass
class AuthMeDto:
    """ログイン中スタッフ情報 Dto。"""

    staff_id: int
    name: str
    avatar: Optional[str]
    role: int

File: backends/ts-hono/src/usecase/auth/dto.ts
/**
 * 認証ユースケース DTO モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** ログイン入力 DTO。 */
export interface LoginInput {
  provider: number;
  providerId: string;
  name: string;
  email: string;
  avatar?: string;
}
